File: HackerRank/30DaysOfCode/CPP/day14_scope.cpp
#include <bits/stdc++.h>
using namespace std;

class Difference
{
private:
    vector<int> elements;

public:
    int maximumDifference;

    Difference(vector<int> arr)
    {
        elements = arr;
        sort(elements.begin(), elements.end());
    }

    void computeDifference()
    {
        maximumDifference = elements[elements.size() - 1] - elements[0];
    }

    void computeDifferenceAlternative()
    {
        int min = elements[0], max = elements[0];
        for (int i = 1; i < elements.size(); i++)
        {
            if (elements[i] > max)
            {
                max = elements[i];
            }
            if (elements[i] < min)
            {
                min = elements[i];
            }
        }
        maximumDifference = max - min;
    }
};

int main()
{
    int n;
    cin >> n;
    vector<int> numbers;
    for (int i = 0; i < n; i++)
    {
        int number;
        cin >> number;
        numbers.push_back(number);
    }
    Difference difference(numbers);
    difference.computeDifference();
    cout << difference.maximumDifference << endl;
    return 0;
}

File: HackerRank/30DaysOfCode/Python/day14_scope.py
class Difference:
    def __init__(self, arr):
        self.__elements = arr
        self.maximum_difference = 0
        
    def compute_difference(self):
        self.maximum_difference = max(self.__elements) - min(self.__elements)
        
        
if __name__ == '__main__':
    _ = input()
    a = [int(e) for e in input().split(' ')]
    d = Difference(a)
    d.compute_difference()
    print(d.maximum_difference)